api/analyzers/csharp/analyzer.py

```
@@ -0,0 +1,144 @@
+
+class CSharpAnalyzer(AbstractAnalyzer):
+    def __init__(self) -> None:
+        super().__init__(Language(tscsharp.language()))
+
+    def _captures(self, pattern: str, node: Node) -> dict:
+        """Run a tree-sitter query and return captures dict."""
+        query = self.language.query(pattern)
+        cursor = QueryCursor(query)
+        return cursor.captures(node)
+
+    def add_dependencies(self, path: Path, files: list[Path]):
```

Ruff (0.15.1)

[warning] Unused method argument: `files`

(ARG002)